src/Sound/Tidal/Control.hs

```diff
@@ -285,7 +285,9 @@ randslice = tParam $ \n p -> innerJoin $ (\i -> _slice n i p) <$> _irand n
 
 _splice :: Int -> Pattern Int -> ControlPattern -> Pattern (Map.Map String Value)
 _splice bits ipat pat = withEvent f (slice (pure bits) ipat pat) # P.unit (pure "c")
-  where f ev = ev {value = Map.insert "speed" (VF d) (value ev)}
+  where f ev = case Map.lookup "speed" (value ev) of
+                        Nothing -> ev {value = Map.insert "speed" (VF d) (value ev)}
```

Comment:
This is causing a warning, it'll fail on a non floating point `speed` value. That's unlikely to happen, but worth swapping the two lines around, and replacing `Nothing` with `_` as the default case.

Reply:
Ah yes that makes sense! Thanks!